processing/src/main/java/io/druid/segment/StringDimensionHandler.java

```diff
@@ -120,24 +120,20 @@ public void validateSortedEncodedArrays(
               dimValName
           );
         }
+      } else {
```

Comment:
Hmm, that's true. I think we can get rid of `if (lhsLen == 0 && rhsLen == 1)` and `else if (rhsLen == 0 && lhsLen == 1)` statements and instead check if there's any null in `lhs` and `rhs` to handle situations such as when events are defined as following:
```
Map<String, Object> event1 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList("", ""),
        "predators", Arrays.asList("seal")
    );
    Map<String, Object> event2 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList(""),
        "predators", Arrays.asList("seal")
    );
```
or 
```
Map<String, Object> event1 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList("", ""),
        "predators", Arrays.asList("seal")
    );
    Map<String, Object> event2 = ImmutableMap.<String, Object>of(
        "penguins", Arrays.asList(),
        "predators", Arrays.asList("seal")
    );
```
I would consider two segments to be same if they produce the same query results, so I think in the above case the validation should pass.